service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenMigrationService.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.PartyType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.ChildParty;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class ChildrenMigrationService {
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    public String setMigratedValue(CaseData caseData) {
+        if (caseData.getChildren1() != null || caseData.getChildren() == null) {
+            return "Yes";
+        } else {
+            return "No";
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Element<Child>> expandChildrenCollection(CaseData caseData) {
+        if (caseData.getChildren1() == null) {
+            List<Element<Child>> populatedChildren = new ArrayList<>();
+
+            populatedChildren.add(Element.<Child>builder()
+                .value(Child.builder()
+                    .party(ChildParty.builder()
+                        .partyId(UUID.randomUUID().toString())
+                        .build())
+                    .build())
+                .build());
+            return populatedChildren;
+        } else {
+            return caseData.getChildren1();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<Element<Child>> addHiddenValues(CaseData caseData) {
+        List<Element<Child>> childrenParties = caseData.getChildren1();
+
+        List<ChildParty> childrenPartyList = childrenParties.stream()
+            .map(Element::getValue)
+            .map(Child::getParty)
+            .map(child -> {
+                ChildParty.ChildPartyBuilder partyBuilder = child.toBuilder();
+
+                if (child.getPartyId() == null) {
+                    partyBuilder.partyId(UUID.randomUUID().toString());
+                    partyBuilder.partyType(PartyType.INDIVIDUAL);
+                }
+
+                return partyBuilder.build();
+            })
+            .collect(toList());
+
+        List<Element<Child>> children = childrenPartyList.stream()
+            .map(item -> Element.<Child>builder()
+                .id(UUID.randomUUID())
+                .value(Child.builder()
+                    .party(item)
+                    .build())
+                .build())
+            .collect(toList());
+
+        return children;
+    }
```

Comment:
This method should give you issues due to the random UUID you are generating everytime the method is called.

I would do something like:

```
caseData.getChildren1().stream
    .map(element -> {
        instantiate child party builder
            logic for adding partyId and partyType
        build element. 
        here you have access to the CCD generated ID with element.getId(), 
        add your child party as the value
        }
    collect to list
    return the List<Element<Child>> you've made 
```